Simplenote/NoteContentHelper.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+// MARK: - NoteContentHelper
+//
+struct NoteContentHelper {
+
+    private init() {
+
+    }
+
+    /// Returns structure of the content: range of title and body
+    ///
+    /// - Parameters:
+    ///     - content: note content
+    ///
+    static func structure(of content: String?) -> (title: NSRange, body: NSRange) {
+        let titleRange = trimmedTextRange(in: content, startingFrom: 0, endAtNewline: true)
+        let bodyRange: NSRange
+        if titleRange.isNotFound {
```

Comment:
Q: How about `NSRange?` (and drop the sentinel?)

Reply:
Do you mean to use and return an optional NSRange? I thought about it but wanted to be consistent with other APIs. From what I see NSRange is never used as an optional value, even in swift.